### PR TITLE
Add templates deployment to kubevirt ansible

### PR DESCRIPTION
**Pull request by yanirq**
**What this PR does / why we need it**:
Add deployment of common templates using kubevirt ansible based on [templates v0.4.0](https://github.com/kubevirt/common-templates/releases/tag/v0.4.0)
The templates included here are for : cenots7, rhel7 and win2k12r2 (this is cut from the common-templates file)

**Special notes for your reviewer**:
This is a temporary deployment solution until we use deployment with ssp-operator (ansible operator sdk)
**Release note**:

```release-note
deploy kubevirt templates 
```

### CODE REVIEW

**Comment by yanirq** [comment]
@MarSik @rthallisey

File: roles/kubevirt-ssp/tasks/provision.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    path: "/tmp/common-templates.yaml"
+  register: common_templates
+
+- name: Copy common templates yaml to temp directory
+  copy:
+    src: "{{ kubevirt_ssp_files_dir }}/common-templates-v0.4.0.yaml"
+    dest: "/tmp/common-templates.yaml"  
+  when: common_templates.stat.exists == false
+
+- name: Create Kubevirt Template
```
Comment (rthallisey) [inline roles/kubevirt-ssp/tasks/provision.yml]: Add a verification step that the operator deployed what was expected before we proceede

File: roles/kubevirt-ssp/tasks/deprovision.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  stat:
+    path: "/tmp/common-templates.yaml"
+  register: common_templates
+
+- name: Copy common templates yaml to temp directory
+  copy:
+    src: "{{ kubevirt_ssp_files_dir }}/common-templates-v0.4.0.yaml"
+    dest: "/tmp/common-templates.yaml"  
+  when: common_templates.stat.exists == false
+
```
Comment (rthallisey) [inline roles/kubevirt-ssp/tasks/deprovision.yml]: Needs a verification step to confirm we have cleaned up everything.

File: roles/kubevirt-ssp/tasks/deprovision.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+- name: Copy common templates yaml to temp directory
+  copy:
+    src: "{{ kubevirt_ssp_files_dir }}/common-templates-v0.4.1.yaml"
+    dest: "/tmp/common-templates.yaml"  
+  when: common_templates.stat.exists == false
+
+- name: Delete Kubevirt Templates
+  shell: "{{ cluster_command }} delete -f /tmp/common-templates.yaml -n {{ kubevirt_templates_namespace }}"
+
+- name: Wait until all templates are deleted
```
[the remark this answers is not in the document]
Comment (MarSik) [reply roles/kubevirt-ssp/tasks/deprovision.yml]: @rthallisey The wait here is not necessary at all, oc delete is atomic. The templates are gone once the command finishes.

File: roles/kubevirt-ssp/tasks/provision.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  shell: "{{ cluster_command }} apply -f /tmp/common-templates.yaml -n {{ kubevirt_templates_namespace }}"
+
+- name: Wait until all templates are created
+  shell: "{{ cluster_command }} -n {{ kubevirt_templates_namespace }} get templates | grep -o -E \"(rhel7-|centos7-|win2k12r2-)\" | wc -l"
+  register: result
+  until: result.stdout == "14"
```
[the remark this answers is not in the document]
Comment (MarSik) [reply roles/kubevirt-ssp/tasks/provision.yml]: The wait here is unnecessary, oc create is atomic. The templates are already created when the command finishes. This kind of wait is only needed for the operator case.

**Pull request merged**